scanpipe/pipes/symbolmap.py

```diff
@@ -36,10 +38,54 @@
 MATCHING_RATIO_RUST = 0.5
 SMALL_FILE_SYMBOLS_THRESHOLD = 20
 MATCHING_RATIO_RUST_SMALL_FILE = 0.4
+MATCHING_RATIO_ELF = 0.05
+STATS_FIELDS = [
+    "matched_symbols",
+    "unmatched_symbols",
+    "matched_symbols_ratio",
+    "matched_symbols_unique_ratio",
+    "source_symbols_count",
+    "matched_symbols_count",
+]
+
+
+@dataclass
+class SymbolsMatchResult:
+    """
+    Statistics and debug information for each attempted matching
+    between a binary and a source file using extracted symbols.
+    Contains CodebaseRelation and other data to create the codebase
+    realtions optionally if matching with symbols is successful.
+    """
+
+    from_resource: str
+    is_matched: bool
+    relation_key: tuple | None = None
```

Comment:
I like very much this style. Between this and pydantic, I am wondering if we should replace attrs.